android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java

```diff
@@ -0,0 +1,392 @@
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+
+public class DSTManager {
+
+    private SimpleDateFormat simpleDateFormat;
+    private AlarmManager alarmMgr;
+    private String DST_STATUS = DST.DST_START.name();
+    //private boolean timeZoneChangeReceiverIsRegistered = false;
+
+
+    public void scheduleDSTTransitionAlarms(Context context) {
+        simpleDateFormat = new SimpleDateFormat("dd/MMM/yyyy HH:mm a", Locale.getDefault());
+        alarmMgr = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        List<DSTData> dstList = checkTimeZoneIsOnDSTAndGetDSTData(context, TimeZone.getDefault().getID());
+        Calendar currentCalendar = Calendar.getInstance();
+        /* Receiver should be registered once*/
+        /*if (!timeZoneChangeReceiverIsRegistered && context instanceof Activity) {
+            registerTimeZoneChangeReceiver(context);
+        }*/
+        int dst_start_request_code = DST.DST_START.getValue();
+        int dst_end_request_code = DST.DST_END.getValue();
+        /* we are using loops because if we have countries with changing dst-start/end date,
+        ie; iran have 21/22 march as dst start date*/
+        for (DSTData dstObject : dstList) {
+            int startMonth = dstObject.getStart_month_index();
+            int endMonth = dstObject.getEnd_month_index();
+            int startTime = dstObject.getStart_time();
+            int endTime = dstObject.getEnd_time();
+            int year = currentCalendar.get(Calendar.YEAR);
+
+            int startDay;
+            int endDay;
+            if (dstObject.getStart_week_day() != null & dstObject.getEnd_week_day() != null) {
+                String startWeekDay = dstObject.getStart_week_day().toUpperCase();
+                String endWeekDay = dstObject.getEnd_week_day().toUpperCase();
+                startDay = getDayFromWeekDay(startWeekDay, startMonth, year);// ie; LSAT_SUNDAY 28
+                endDay = getDayFromWeekDay(endWeekDay, startMonth, year);// ie; LAST_SUNDAY 28
+            } else {
+                startDay = dstObject.getStart_day();
+                endDay = dstObject.getEnd_day();
+            }
+
+            Calendar startCalendar = Calendar.getInstance();
+            startCalendar.set(Calendar.MONTH, startMonth);
+            startCalendar.set(Calendar.DAY_OF_MONTH, startDay);
+            startCalendar.set(Calendar.HOUR_OF_DAY, startTime);
+            startCalendar.set(Calendar.MINUTE, 0);
+            startCalendar.set(Calendar.SECOND, 0);
+            startCalendar.set(Calendar.MILLISECOND, 0);
+
+            Calendar endCalendar = Calendar.getInstance();
+            endCalendar.set(Calendar.MONTH, endMonth);
+            endCalendar.set(Calendar.DAY_OF_MONTH, endDay);
+            endCalendar.set(Calendar.HOUR_OF_DAY, endTime);
+            endCalendar.set(Calendar.MINUTE, 0);
```

Comment:
We should include minutes along with hour.

Reply:
Ok noted.

Reply:
Those are from Database value. The database only contains hour value.